### DISCUSSION TITLE
Building Using Linux

### DISCUSSION THREAD

**Discussion by tataDan** [raguay/EmailIt#17] [Q&A]
In the README.md the maintainer states :

> I've only built it as a macOS universal build.

Does this mean that the application, as is,  cannot be built with Linux?

I am using the following:

node: 18.16.0
npm: 9.5.1
go: go1.20 linux/amd64
wails: 2.4.0

When I attempt to build the application, I get the following error:
```
Could not resolve "../components/SimpleAutoComplete.svelte" from "src/components/ScriptsEditor.svelte"
    file: /home/daniel/testing/wails-apps/EmailIt/frontend/src/components/ScriptsEditor.svelte
    error during build:
    RollupError: Could not resolve "../components/SimpleAutoComplete.svelte" from "src/components/ScriptsEditor.svelte"
        at error (file:///home/daniel/testing/wails-apps/EmailIt/frontend/node_modules/rollup/dist/es/shared/node-entry.js:2105:30)
        at ModuleLoader.handleInvalidResolvedId (file:///home/daniel/testing/wails-apps/EmailIt/frontend/node_modules/rollup/dist/es/shared/node-entry.js:23630:24)
        at file:///home/daniel/testing/wails-apps/EmailIt/frontend/node_modules/rollup/dist/es/shared/node-entry.js:23592:26
    
  ERROR   exit status 1
```

**Comment by raguay** [comment]
It does build on Linux for me (I've built it on Ubuntu latest build in a virtual machine), but I haven't tested all the features. There are a few items (like launching an editor using the macOS open command) that I haven't changed to work with other OSes. Your error says that it can't find a component that should be in the repo. I assume you have an incomplete download of the project. You might try to reclone the repo, goto the `frontend` directory and run `npm i` and then `npm run build` and see if it builds.

**Reply by raguay** [reply]
I've encountered this error now and I'm looking into it.

**Comment by raguay** [comment]
Hi, if you do a `git pull`, or do a new clone, it should now work. I've successfully built it on the latest Debian linux distro for arm64. Therefore, it should work for you. You will need to update the libraries by running `cd frontend; npm i` and then update Wails with `wails update`. Then you can run `wails build` to make the program on your system.
